src/types/Course/Preference.ts

```diff
@@ -23,8 +24,37 @@ export const PreferenceQuery = extendType({
   definition(t) {
     t.nonNull.field('survey', {
       type: TeachingPreferenceSurvey,
-      description: 'Get Teaching Preference Survey',
-      resolve: () => ({ courses: [] }),
-    });
-  },
+      description: 'Get Teaching Preference Survey for a given User ID',
+      args: {
+        userid: arg({ type: 'Int' }),
+      },
+      resolve: async (_, args, ctx) => {
+        const { userid } = args;
+        const { prisma } = ctx;
+        if(!userid) {
+          return null;
+        }
+        let prefs: any = [];
+        const pref = await (prisma as PrismaClient).preference.findMany({
+          where: {
+            user: {
+              id: userid,
+            }
+          },
+        });
+        
+        for (const p of pref) {
+          prefs.push({
+            id: p.courseID,
+            preference: p.rank,
+          });
+        }
+        return {
+          courses: prefs,
+        };
+          
```

Comment:
```suggestion
        const prefs = await (prisma as PrismaClient).preference.findMany({
          where: {
            user: {
              id: userid,
            }
          },
        });
        
       const courses = prefs.map(({ courseID, rank }) => ({
          id: courseID,
          preference: rank
       }));
       
        return {
          courses,
        };
```
I might have mucked up some syntax cause I wrote this in github but I think this should work the same

Reply:
Beautiful